java/src/test/android/app/build.gradle

```diff
@@ -25,28 +25,28 @@ android {
 		}
 	}
 	compileOptions {
-		sourceCompatibility JavaVersion.VERSION_1_8
-		targetCompatibility JavaVersion.VERSION_1_8
+		sourceCompatibility JavaVersion.VERSION_1_17
```

Comment:
SDK version 33 seems to indicate it's based on OpenJDK 11 - https://developer.android.com/about/versions/13/features#core-libraries, and you need SDK 34 for OpenJDK 17 - https://developer.android.com/about/versions/14/features#core-libraries.

If the baseline is SDK 33, can we move the whole Java API to Java 11 (which is the newest one supported by Android 13)? Being able to use `var` and package things as a proper JPMS module will make a few things simpler for me, and lots of the Java ecosystem has moved to 17 now.

Reply:
I had hard time to upgrade to SDK 34 from previous PR, but we can give it a try again. Thanks for point it out the required SDK version for Java 17.